pallets/ddc-staking/src/lib.rs

```diff
@@ -737,9 +840,140 @@ pub mod pallet {
+			}
+
+			// ToDo: check that all accounts had an active stake at the era
+
+			Self::reward_by_ids(era, stakers_points);
+
+			Ok(())
+		}
+
+		/// Set price per byte of the bucket traffic in smallest units of the currency.
+		///
+		/// The dispatch origin for this call must be _Root_.
+		#[pallet::weight(10_000)]
+		pub fn set_pricing(origin: OriginFor<T>, price_per_byte: u128) -> DispatchResult {
+			ensure_root(origin)?;
+			<Pricing<T>>::set(Some(price_per_byte));
+			Ok(())
+		}
 	}
 
 	impl<T: Config> Pallet<T> {
+		pub fn do_payout_stakers(era: EraIndex) -> DispatchResult {
+			// ToDo: check that the era is finished
+			// ToDo: check reward points are set
+
+			let era_reward_points: EraRewardPoints<T::AccountId> =
+				<ErasEdgesRewardPoints<T>>::get(&era);
+
+			let price_per_byte: u128 = match Self::pricing() {
+				Some(pricing) => pricing,
+				None => Err(Error::<T>::PricingNotSet)?,
+			};
+
+			// An account we withdraw the funds from and the amount of funds to withdraw.
+			let payout_source_account: T::AccountId =
+				T::StakersPayoutSource::get().into_account_truncating();
+			let payout_budget: BalanceOf<T> =
+				match (price_per_byte * era_reward_points.total as u128).try_into() {
+					Ok(value) => value,
+					Err(_) => Err(Error::<T>::BudgetOverflow)?,
+				};
+			log::info!(
+				"Will payout to DDC stakers for era {:?} from account {:?} with total budget {:?} \
+				, there are {:?} stakers earned {:?} reward points with price per byte {:?}",
+				era,
+				payout_source_account,
+				payout_budget,
+				era_reward_points.individual.len(),
+				era_reward_points.total,
+				price_per_byte,
+			);
+
+			// Transfer a part of the budget to each CDN participant rewarded this era.
+			for (stash, points) in era_reward_points.clone().individual {
+				let part = Perbill::from_rational(points, era_reward_points.total);
+				let reward: BalanceOf<T> = part * payout_budget;
+				log::info!(
+					"Rewarding {:?} with {:?} points, its part is {:?}, reward size {:?}, balance \
+					on payout source account {:?}",
+					stash,
+					points,
+					part,
+					reward,
+					T::Currency::free_balance(&payout_source_account)
+				);
+				T::Currency::transfer(
+					&payout_source_account,
+					&stash,
+					reward,
+					ExistenceRequirement::AllowDeath,
+				)?; // ToDo: all success or noop
+				Rewards::<T>::mutate(&stash, |current_balance| {
+					*current_balance += reward;
+				});
+				log::info!("Total rewards to be inserted: {:?}", Self::rewards(&stash));
+				PaidErasPerNode::<T>::mutate(&stash, |current_rewards| {
+					let rewards = EraRewardsPaid { era, reward };
+					current_rewards.push(rewards);
+				});
+			}
+			Self::deposit_event(Event::<T>::PayoutNodes(
+				era,
+				era_reward_points.clone(),
+				price_per_byte,
+			));
+			log::info!("Payout event executed");
+
+			log::debug!(
+				"Balance left on payout source account {:?}",
+				T::Currency::free_balance(&payout_source_account),
```

Comment:
is there a case where there is an imbalance between content owner deposits and payouts?

Reply:
We don't have content owners here, but no imbalance between the `payout_source_account` and stakers accounts is guaranteed by the `T::Currency::transfer`.